src/script/interpreter.cpp

```diff
@@ -1832,6 +1832,10 @@ uint256 ComputeTapleafHash(uint8_t leaf_version, const CScript& script)
 
 uint256 ComputeTaprootMerkleRoot(Span<const unsigned char> control, const uint256& tapleaf_hash)
 {
+    assert(control.size() >= TAPROOT_CONTROL_BASE_SIZE
+           && control.size() <= TAPROOT_CONTROL_MAX_SIZE
+           && ((control.size() - TAPROOT_CONTROL_BASE_SIZE) % TAPROOT_CONTROL_NODE_SIZE == 0));
```

Comment:
nit: Probably it would make sense to split those three conditions (too small, too large, length not 33 + 32m) up, so in the  case that this ever throws, we can yield more information about the size of the control block:
```suggestion
    assert(control.size() >= TAPROOT_CONTROL_BASE_SIZE);
    assert(control.size() <= TAPROOT_CONTROL_MAX_SIZE);
    assert((control.size() - TAPROOT_CONTROL_BASE_SIZE) % TAPROOT_CONTROL_NODE_SIZE == 0);
```

Reply:
Applied suggestion, thanks.